internal/controller/pod_webhook.go

```diff
@@ -45,11 +46,40 @@ func (a *PodAnnotator) Handle(ctx context.Context, req admission.Request) admiss
 	if err != nil {
 		return admission.Errored(400, fmt.Errorf("could not decode pod: %v", err))
 	}
-
+	log.FromContext(ctx).Info("webhook got pod ", "name", pod.Name)
 	if !hasMIGResource(pod) {
 		return admission.Allowed("No nvidia.com/mig-* resource found, skipping mutation.")
-	} else {
-		performQuotaArithmetic(pod, req)
+	}
+
+	performQuotaArithmetic(pod, req)
+
+	// Add extended resource to resource limits
+	if pod.Spec.Containers[0].Resources.Limits == nil {
+		pod.Spec.Containers[0].Resources.Limits = make(v1.ResourceList)
+	}
+
+	limits := pod.Spec.Containers[0].Resources.Limits
+	for resourceName, quantity := range limits {
+		if strings.HasPrefix(string(resourceName), "nvidia.com/mig-") {
+			newResourceName := strings.Replace(string(resourceName), "nvidia.com", "org.instaslice", 1)
+			delete(pod.Spec.Containers[0].Resources.Limits, resourceName)
+			limits[v1.ResourceName(newResourceName)] = quantity
+		}
+	}
+
+	if pod.Spec.Containers[0].Resources.Requests == nil {
+		log.FromContext(ctx).Info("Initializing resource requests for ", "pod", pod.Name)
```

Comment:
should be a debug

Reply:
good catch, I will remove that logger, this was for debugging